sympy/solvers/solvers.py

```diff
@@ -2718,6 +2718,22 @@ def equal(expr1, expr2):
         # lambert forms may need some help being recognized, e.g. changing
         # 2**(3*x) + x**3*log(2)**3 + 3*x**2*log(2)**2 + 3*x*log(2) + 1
         # to 2**(3*x) + (x*log(2) + 1)**3
+        logs = eq.atoms(log)
+        spow = {i.exp for j in logs for i in j.atoms(Pow) if i.base == sym}
+        spow = min(spow) if spow else 1
+        if eq.subs(sym**spow, Dummy()).has_free(sym):
+            spow = 1  # there was a free-symbol or non-pow generator with sym
+        if spow != 1:
+            u = Dummy('bivariate-cov')
+            p = sym**spow
+            ueq = eq.subs(p, u)
```

Comment:
```suggestion
        p = sym**spow
        u = Dummy('bivariate-cov')
        ueq = eq.subs(p, u)
        if not ueq.has(sym):
```

Reply:
But extracting the `min`, even though `spow > 1` is not used, will fail if any power has a symbolic exponent. I can give a better test later if you don't come up with one.

Reply:
How about `solve((1/log(x**(2*y) + 2)**2 - x**-(4*y)))`?

Reply:
What is the value of the `spow` **set** when solving that equation?